packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -0,0 +1,158 @@
+final PlatformViewRegistry platformViewRegistry = PlatformViewRegistry._instance();
+
+/// A registry responsible for generating unique identifier for platform views.
+///
+/// A Flutter application has a single [PlatformViewRegistry] which can be accesses
+/// through the [platformViewRegistry] getter.
+///
+/// See also:
+///   * [PlatformView], a widget that shows a platform view.
+class PlatformViewRegistry {
+  PlatformViewRegistry._instance();
+
+  int _nextPlatformViewId = 0;
+
+  /// Allocates a unique identifier for a platform view.
+  ///
+  /// A platform view identifier can refer to a platform view that was never created,
+  /// a platform view that was disposed, or a platform view that is alive.
+  ///
+  /// Typically a platform view identifier is passed to a [PlatformView] widget
+  /// which creates the platform view and manages its lifecycle.
+  int getNextPlatformViewId() => _nextPlatformViewId++;
+}
+
+/// Callback signature for when a platform view was created.
+///
+/// `id` is the platform view's unique identifier.
+typedef void OnPlatformViewCreated(int id);
+
+/// Provides access to the platform views service.
+///
+/// This service allows creating and controlling Android views.
+///
+/// See also: [PlatformView].
+class PlatformViewsService {
+  PlatformViewsService._();
+
+  /// Creates a controller for a new Android view.
+  ///
+  /// `id` is an unused unique identifier generated with [platformViewsRegistry].
+  ///
+  /// `viewType` is the identifier of the Android view type to be created, a
+  /// factory for this view type must have been registered on the platform side.
+  /// Platform view factories are typically registered by plugin code.
+  /// Plugins can register a platform view factory with
+  /// [PlatformViewRegistry#registerViewFactory](/javadoc/io/flutter/plugin/platform/PlatformViewRegistry.html#registerViewFactory-java.lang.String-io.flutter.plugin.platform.PlatformViewFactory-).
+  ///
+  /// The Android view will only be created after [AndroidViewController.setSize] is called for the
+  /// first time.
+  static AndroidViewController initAndroidView({
+    @required int id,
+    @required String viewType,
+    OnPlatformViewCreated onPlatformViewCreated,
+  }) {
+    return new AndroidViewController._(
+        id: id,
+        viewType: viewType,
+        onPlatformViewCreated: onPlatformViewCreated
+    );
+  }
+}
+
+enum _AndroidViewState {
+  WaitingForSize,
+  Creating,
+  Created,
+  CreateFailed,
+  Disposed,
+}
+
+/// Controls an Android view.
+///
+/// Typically created with [PlatformViewsService.initAndroidView].
+class AndroidViewController {
+  AndroidViewController._({
+    @required this.id,
+    @required String viewType,
+    OnPlatformViewCreated onPlatformViewCreated,
+  }) : assert(id != null),
+       assert(viewType != null),
+       _viewType = viewType,
+        _onPlatformViewCreated = onPlatformViewCreated,
+       _state = _AndroidViewState.WaitingForSize;
+
```

Comment:
https://www.dartlang.org/guides/language/effective-dart/usage#do-initialize-fields-at-their-declaration-when-possible

For this and the other instance